components/lock.rst

```diff
@@ -0,0 +1,319 @@
+.. index::
+   single: Lock
+   single: Components; Lock
+
+The Lock Component
+====================
+
+    The Lock Component provides a mechanism to garentee an exclusive access into
+    a critical section. The component ships with ready to use stores for the
+    most common backends.
+
+.. versionadded:: 3.3
+    The Lock component was introduced in Symfony 3.3.
+
+Installation
+------------
+
+You can install the component in 2 different ways:
+
+* :doc:`Install it via Composer </components/using_components>` (``symfony/lock`` on `Packagist`_);
+* Use the official Git repository (https://github.com/symfony/lock).
+
+.. include:: /components/require_autoload.rst.inc
+
+
+Usage
+-----
+
+In order to centralize state of locks, you first need to create a ``Store``.
+Then, you can ask to this store to create a Lock for your ``resource``.
+
+The :method:`Symfony\\Component\\Lock\\LockInterface::acquire` method tries to
+acquire the lock. If the lock is can not be acquired, the method throws a
+:class:`Symfony\\Component\\Lock\\Exception\\LockConflictedException`. You can
+safly call the ``acquire()`` method several time, even if you already acquired
+it.
+
+.. code-block:: php
+
+    use Symfony\Component\Lock\Store\SemaphoreStore;
+    use Symfony\Component\Lock\Exception\LockConflictedException;
+
+    $store = new SemaphoreStore();
+    $lock = $store->createLock('hello');
```

Comment:
This component may be tricky to understand to some people, so I recomment to use realistic code samples (in other words, replace `hello`, `foo`, `bar` with real-world looking examples).